test/mock/makeMockUseValidate.ts

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { match } from 'ts-pattern'
+
+import { ValidationResult } from '@app/hooks/useValidate'
+
+export const mockUseValidateConfig = {
+  eth: { input: 'eth' },
+  dns: { input: 'com' },
+  'valid-2ld': { input: 'name.eth' },
+  'valid-2ld:dns': { input: 'name.com' },
+  'invalid-2ld': { input: 'name❤️.eth' },
+  'valid-subname': { input: 'subname.name.eth' },
+} as const
+export type MockUseValidateType = keyof typeof mockUseValidateConfig
+export const mockUseValidateTypes = Object.keys(mockUseValidateConfig) as MockUseValidateType[]
+
+export const makeMockUseValidate = (type: MockUseValidateType): ValidationResult => {
```

Comment:
does useValidate need to be mocked

Reply:
In the absence of tests for the individual functions, imo I would say it useful as a test to make sure that the results remain consistent with what we expect in case we modify any of those functions in the future.